src/CP/valueselection/rewards/defaultreward.jl

```diff
@@ -14,11 +14,22 @@ DefaultReward(model::CPModel) = DefaultReward(0)
 
 Change the "current_reward" attribute of the LearnedHeuristic at the StepPhase.
 """
-function set_reward!(::StepPhase, lh::LearnedHeuristic{SR, DefaultReward, A}, model::CPModel, symbol::Union{Nothing, Symbol}) where {
+function set_reward!(::Type{StepPhase}, lh::LearnedHeuristic{SR, DefaultReward, A}, model::CPModel, symbol::Union{Nothing, Symbol}) where {
     SR <: AbstractStateRepresentation,
     A <: ActionOutput
 }
-    nothing
+    if symbol == :Infeasible  
+        lh.reward.value -= 10*model.statistics.numberOfNodes
+
+    elseif symbol == :FoundSolution
+        lh.reward.value -= 1
+
+    elseif symbol == :Feasible 
+        lh.reward.value -= 1
+
+    elseif symbol == :BackTracking
+        lh.reward.value -= 1
+    end
```

Comment:
Finally we don't need to declare another function : `function set_reward!(::SolutionFoundPhase, ... ` as the information about the search (` :SolutionFound`, `:Feasible`, `:BackTracking `) is already contained in the argument `symbol`. 

We simply need to inform the user that he could implement a reward that can consider different symbols.